compiler/qsc_rca/src/applications.rs

```diff
@@ -0,0 +1,666 @@
+            {
+                Self::aggregate_param_application_value_kind(
+                    param_application,
+                    &compute_properties,
+                );
+            }
+
+            // Return the applications gene with the updated dynamism sources.
+            applications_generator
+        });
+
+        close_output
+    }
+
+    fn aggregate_param_application_value_kind(
+        param_application: &mut crate::ParamApplication,
+        compute_properties: &ParamApplicationComputeProperties,
+    ) {
+        match param_application {
+            crate::ParamApplication::Array(array_param_application) => {
+                let ParamApplicationComputeProperties::Array(array_compute_properties) =
+                    compute_properties
+                else {
+                    panic!("expected an array param application");
+                };
+                array_compute_properties
+                    .static_content_dynamic_size
+                    .value_kind
+                    .iter()
+                    .for_each(|value_kind| {
+                        array_param_application
+                            .static_content_dynamic_size
+                            .aggregate_value_kind(*value_kind);
+                    });
+                array_compute_properties
+                    .dynamic_content_static_size
+                    .value_kind
+                    .iter()
+                    .for_each(|value_kind| {
+                        array_param_application
+                            .dynamic_content_static_size
+                            .aggregate_value_kind(*value_kind);
+                    });
+                array_compute_properties
+                    .dynamic_content_dynamic_size
+                    .value_kind
+                    .iter()
+                    .for_each(|value_kind| {
+                        array_param_application
+                            .dynamic_content_dynamic_size
+                            .aggregate_value_kind(*value_kind);
+                    });
+            }
+            crate::ParamApplication::Element(element_param_application) => {
+                let ParamApplicationComputeProperties::Element(element_compute_properties) =
+                    compute_properties
+                else {
+                    panic!("expected an element param application");
+                };
+                element_compute_properties
+                    .value_kind
+                    .iter()
+                    .for_each(|value_kind| {
+                        element_param_application.aggregate_value_kind(*value_kind);
+                    });
+            }
+        };
+    }
+
+    fn close_inherent(&mut self) -> ApplicationInstanceComputeProperties {
+        // The inherent param application is always the first one.
+        let mut variants = self.application_instances[0].drain(..).collect_vec();
```

Comment:
This can use standard library for this rather than `collect_vec` like this:

```suggestion
        let mut variants = self.application_instances[0].drain(..).collect::<Vec<_>>();
```